### PR TITLE
Simplify requirements*.txt

### PR DESCRIPTION
**Pull request by janiversen**
Cleaned installation procedure:

requirements.txt contains all dependencies, mainly because it is only used by people who clone the repo, and thus are very likely to be developers who need all dependencies (or is capable of adjusting to Specific needs).

Setup.cfg contain all fixed setup information.

Setup.py reads requirements.txt to generate dependencies as either fixed for opcional (allowing Pymodbus[<option>]

README.rst have been updated to reflect the above changes.

### CODE REVIEW

**Comment by janiversen** [comment]
The CI errors are not caused by this PR, but probably somewhere in managing a stack of 6 PR on top of each other (waiting for approval).

Once approved I will rebase each PR and let CI verify.

File: requirements.txt
Comment (dhoomakethu) [inline requirements.txt:30]: We do not have to install these modules unless users explicitly wants to use these features. For e.g if somebody wants to install pymodbus with `REPL` they can simply to `pip install pymodbus[repl]`.

Comment (dhoomakethu) [reply requirements.txt:30]: Refer setup.py

File: requirements-tests.txt
Comment (dhoomakethu) [inline requirements-tests.txt:20]: We can also upgrade pytest to latest (6.x and above)

File: requirements.txt
Comment (janiversen) [reply requirements.txt:30]: I agree it is not a demand, but it makes everything a lot easier, and what is the drawback...they occupy a bit of space in "site-packages". I wrote in the text that they can be commented out.

File: requirements-tests.txt
Comment (dhoomakethu) [inline requirements-tests.txt:23]: I remember there was an issue with `asynctest` for some versions of pytest. Will have to check that out.

Comment (janiversen) [reply requirements-tests.txt:23]: ok

File: requirements.txt
Comment (dhoomakethu) [reply requirements.txt:30]: I am ok with commented out `requirements.txt` for easy reference except for pyserial and pyserial-async packages

Comment (janiversen) [reply requirements.txt:30]: My preference was to go the other way, install all unless the end-user comments it out. The reason is that many users to not view the content of the files but simply install them.

File: requirements-tests.txt
Comment (dhoomakethu) [reply requirements-tests.txt:23]: Refer https://github.com/riptideio/pymodbus/issues/681

Comment (swamper123) [reply requirements-tests.txt:23]: In the actual versions of python, unittest support async stuff.
I would either recommened to switch to [builtin stuff](https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.IsolatedAsyncioTestCase) or as an alternative use the pytest plugin "[pytest-asyncio](https://github.com/pytest-dev/pytest-asyncio)".

['asynctest'  seems to be abandoned (last commit 2019)](https://github.com/Martiusweb/asynctest)

Comment (janiversen) [reply requirements-tests.txt:23]: That is a good hint !!!

I will take a look later and see how easy it is to remove, stay tuned.

Comment (janiversen) [reply requirements-tests.txt:23]: I think I am able to remove asynctest, I will do that before this one gets merged.

@dhoomakethu I removed package "attrs", because I could not find any references, but I just saw a commit in your "black" branch that upgrade the version, have I missed something ?

Comment (dhoomakethu) [reply requirements-tests.txt:23]: I do not remember using `attrs` any where explicitly, I believe it came in as part of the `black` package dependency.

Comment (janiversen) [reply requirements-tests.txt:23]: thanks, yes it is a requisite for black, but we should not list the dependencies for other packages.

**Comment by janiversen** [comment]
It turned out removing asynctest was not as easy as I thought (pytest structure is better than unittest!!!), so I left that for now.

I have updated all test/check packages to their newest version (including for REPL!). My preference is to do it in a "bump" PR, but...

We still need to update
- click
- prompt-toolkit
- SQLAlchemy
- redis
but they seem to demand code changes.

**Comment by janiversen** [comment]
@dhoomakethu, @swamper123 
I have now remate this PR, please see PR-text and README.rst

This is a simpler way of handling all the different needs, and ensuring dependency versions are kept identical (Currently different Versions are used in the different setup files).

Please have a look, and let me know what you Think.

File: README.rst
Comment (swamper123) [inline README.rst:125]: Better:

Also, if you have a question, please [create an post in discussions q&a topic](https://github.com/riptideio/pymodbus/discussions/new?category=q-a), so that others can benefit from the results [...].

If you think, that something in the code is broken/not running well, please [open an issue](https://github.com/riptideio/pymodbus/issues/new), read the Template-text first and then post your issue with your setup informations.

File: setup.cfg
Comment (swamper123) [inline setup.cfg:19]: I am not so deep in that Licence thing....do we have a BSD Licence? What kind of Licence is this one https://github.com/riptideio/pymodbus/blob/dev/LICENSE ? @dhoomakethu

Comment (swamper123) [reply setup.cfg:19]: Just saw, that this is the old state as well...

**Review by swamper123 — CHANGES_REQUESTED** [review]
We should add the difference between discussion and issue here with a link to avoid mixing up both.
My comment about  the license thing is obsolete .. saw it to late that this seems to be utd.

File: README.rst
Comment (janiversen) [reply README.rst]: Done.

**Review by swamper123 — APPROVED** [review]

**Comment by dhoomakethu** [comment]
Looks good to me.

**Review by dhoomakethu — APPROVED** [review]

**Pull request merged**
